File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/inmemory/SurveyAnswersRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.SurveyAnswersRepository;
import java.math.*;
import java.time.*;
import java.util.*;

public class SurveyAnswersRepositoryInMemory extends InMemoryJpaRepository<SurveyAnswers> implements SurveyAnswersRepository {
    @Override
    public java.util.Optional<SurveyAnswers> findBySurveyIdAndPatientIdAndDate(Long surveyId, Long patientId, LocalDate date) {
        return getEntities().values().stream().filter(e ->
             isSameValue(surveyId, readField(e, "surveyId")) && isSameValue(patientId, readField(e, "patientId")) && isSameValue(date, readField(e, "date")) 
        ).findFirst();
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/inmemory/SurveyAnswersRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.SurveyAnswersRepository
import java.math.*
import java.time.*
import java.util.*

class SurveyAnswersRepositoryInMemory : InMemoryJpaRepository<SurveyAnswers, Long>(), SurveyAnswersRepository {
    override fun findBySurveyIdAndPatientIdAndDate(surveyId: Long, patientId: Long, date: LocalDate): SurveyAnswers? {
        return getEntities().values.firstOrNull { e ->
            isSameValue(surveyId, readField(e, "surveyId")) &&
                isSameValue(patientId, readField(e, "patientId")) &&
                isSameValue(date, readField(e, "date"))
        }
    }
}
